source/_docs/automation/templating.markdown

```diff
@@ -44,9 +44,10 @@ The following describes trigger data associated with all platforms.
 | Template variable | Data |
 | ---- | ---- |
 | `trigger.platform` | Hardcoded: `device`.
-| `trigger.event` | Event object that matched.
-| `trigger.event.event_type` | Event type.
-| `trigger.event.data` | Optional event data.
+| `trigger.entity_id` | Entity ID that we observe.
+| `trigger.from_state` | The previous [state object] of the entity.
+| `trigger.to_state` | The new [state object] that triggered trigger.
+| `trigger.for` | Timedelta object how long state has been to state, if any.
```

Comment:
These are the properties for an entity trigger, not device trigger?
Could you point me to a resource that justifies this change?

../Frenck

Reply:
I discovered actual variables by debugging trigger data to log, but I found reference of device trigger providing state variables and not event variables in [this](https://github.com/home-assistant/core/blob/e39672078d259e4366a409f6258f6a0ef61fb58a/homeassistant/components/binary_sensor/device_trigger.py#L273) and [this](https://github.com/home-assistant/core/blob/e39672078d259e4366a409f6258f6a0ef61fb58a/homeassistant/components/binary_sensor/device_trigger.py#L303) pieces of code. According to it, device trigger has same attributes as entity trigger with according domain.

Simple automation yaml for debugging device trigger data below:
```
alias: Debug device trigger
description: ""
trigger:
  - type: turned_on
    platform: device
    device_id: 03e1f433e4f54fb4695bbb22f96dbb62
    entity_id: binary_sensor.my_phone_interactive
    domain: binary_sensor
condition: []
action:
  - service: system_log.write
    data:
      level: error
      message: "{{ trigger }}"
mode: single
```
And log output example is:
```
{
	'id': '0',
	'idx': '0',
	'alias': None,
	'platform': 'device',
	'entity_id': 'binary_sensor.my_phone_interactive',
	'from_state': < state binary_sensor.my_phone_interactive = off;icon = mdi: cellphone - off,
	friendly_name = my_phone Interactive @ 2022 - 10 - 03 T19: 19: 49.540324 + 04: 00 > ,
	'to_state': < state binary_sensor.my_phone_interactive = on;icon = mdi: cellphone,
	friendly_name = my_phone Interactive @ 2022 - 10 - 03 T19: 36: 44.867720 + 04: 00 > ,
	'for': None,
	'attribute': None,
	'description': 'state of binary_sensor.my_phone_interactive'
}
```

Reply:
The provided links to the code base, are not relevant, as they provide either trigger configuration, not anything related to the firing of events.

Anyways, this might hold true for a binary sensor, as additional information is provided, by it does hold true for others. Device automation is mainly based on/created to help with events. E.g., button presses and similar things.

For example, ZHA button press events on remotes, or things like device automation in zwavejs (and many other places). These will not provide these as listed in your suggested edit.

Please note, while a device config, the provided trace parts, are a state change of an entity.

Reply:
Commited changes according to suggestion above.